DakshChan/IBS
# [IBS-29] Add roles from admin panel

## Overview
- Revamp/Re-create the admin course page such that each view (add, update, upload, delete) is separated by its own tab in the tab-list component.
- Implement the add (assign) roles for users component

### Additional Notes
- On the right-hand side, there is space for notes for the admin to be aware of (so that they know how their actions impact the system)
- The terminology used "add" has been renamed to "assign" as one *assigns* roles to users.

## Testing
- Build and run the docker containers.
- Login as admin, and access an existing course by clicking on its respective card component. This should redirect the admin to the course (admin) page.

## Examples
<img width="1710" alt="Screenshot 2023-06-25 at 8 42 52 PM" src="https://github.com/DakshChan/IBS/assets/47423409/6b9e6fe5-9df6-4d3b-9f5b-190fd90ea731">

## Update [2023-06-27]
Turns out there were a few bugs, which have been addressed in both `[IBS-31]` branch as well as this branch in commit ID [9fcf83d](https://github.com/DakshChan/IBS/pull/21/commits/9fcf83da2ee2dead9c49a15b280ae04aedb8a7e1).
